tests/unit/test_step.py

```diff
@@ -1276,6 +1332,536 @@ def test_embarrassingly_parallel_step__validation(
+                    ],
+                ),
+                Step(
+                    "step_2",
+                    input_slots=[
+                        InputSlot(
+                            "step_2_main_input",
+                            "DUMMY_CONTAINER_MAIN_INPUT_FILE_PATHS",
+                            validate_input_file_dummy,
+                        ),
+                        InputSlot(
+                            "step_2_secondary_input",
+                            "DUMMY_CONTAINER_SECONDARY_INPUT_FILE_PATHS",
+                            validate_input_file_dummy,
+                        ),
+                    ],
+                    output_slots=[
+                        OutputSlot("step_2_main_output"),
+                        OutputSlot("step_2_secondary_output"),
+                    ],
+                ),
+                Step(
+                    "step_3",
+                    input_slots=[
+                        InputSlot(
+                            "step_3_main_input",
+                            "DUMMY_CONTAINER_MAIN_INPUT_FILE_PATHS",
+                            validate_input_file_dummy,
+                        ),
+                        InputSlot(
+                            "step_3_secondary_input",
+                            "DUMMY_CONTAINER_SECONDARY_INPUT_FILE_PATHS",
+                            validate_input_file_dummy,
+                        ),
+                    ],
+                    output_slots=[
+                        OutputSlot("step_3_main_output"),
+                        OutputSlot("step_3_secondary_output"),
+                    ],
+                ),
+            ],
+            edges=[
+                EdgeParams("step_1", "step_2", "step_1_main_output", "step_2_main_input"),
+                EdgeParams(
+                    "step_1", "step_2", "step_1_secondary_output", "step_2_secondary_input"
+                ),
+                EdgeParams("step_2", "step_3", "step_2_main_output", "step_3_main_input"),
+                EdgeParams(
+                    "step_2", "step_3", "step_2_secondary_output", "step_3_secondary_input"
+                ),
+            ],
+            input_slot_mappings=[
+                InputSlotMapping("steps_1_2_3_main_input", "step_1", "step_1_main_input"),
+                InputSlotMapping(
+                    "steps_1_2_3_secondary_input", "step_1", "step_1_secondary_input"
+                ),
+            ],
+            output_slot_mappings=[
+                OutputSlotMapping("steps_1_2_3_main_output", "step_3", "step_3_main_output"),
+                OutputSlotMapping(
+                    "steps_1_2_3_secondary_output", "step_3", "step_3_secondary_output"
+                ),
+            ],
+        ),
+        "input_slots": [
+            InputSlot(
+                "ep_steps_1_2_3_main_input",
+                "DUMMY_CONTAINER_MAIN_INPUT_FILE_PATHS",
+                validate_input_file_dummy,
+                split_data_by_size,
```

Comment:
Maybe just me, but I don't love passing these arguments without keywords (by position)

Reply:
I tend to agree. My carpal tunnel doesn't 😆